### PR TITLE
Suggestions for State pattern.

### PR DESCRIPTION
**Pull request by jolkdarr**
Hello
First, thanks for the project.
I suggest few modifications:
- removal of nested class imports
- code simplifications
- comment about the "empty class" `Unauthorized`

### CODE REVIEW

**Review by dbacinski — COMMENTED** [review]
LGTM, please also update readme.md

**Comment by jolkdarr** [comment]
Done.
Maybe, someone should provide a script able to perform the chore! ;)

File: README.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ Behavioral
 [Observer / Listener](/src/main/kotlin/Listener.kt)
 --------
 
-The observer pattern is used to allow an object to publish changes to its state.
```
Comment (jolkdarr) [inline README.md:39]: My editor automatically dump trailing spaces.

**Review by dbacinski — APPROVED** [review]
Thank you for your contribution.

**Pull request merged**